draft-ietf-quic-recovery.md

```diff
@@ -1129,6 +1129,10 @@ follows:
    smoothed_rtt = kInitialRtt
    rttvar = kInitialRtt / 2
    min_rtt = 0
+   // max_ack_delay is initialized to 0 and
+   // updated when transport parameters are exchanged.
+   // If a peer does not specify max_ack_delay,
+   // it is set to 25ms.
```

Comment:
These comments look fine, but I am not sure if I agree with the rest of the changes.

IIUC, persistent congestion is a path-wide event rather than a event specific to the PN space. I think we need to have max_ack_delay even if that might turn out to be a bit conservative when 0-RTT / 1-RTT packets are not involved.

Then, there is no point in passing `pn_space` to `InPersistentCongestion` or to the functions that invoke `InPersistentCongestion`.

Reply:
My purpose is to make pseudocode consistent with the spec. It seems to me that the spec should be changed if your observation is correct. I'm happy to drop the commit when the spec is changed.

Reply:
I'd argue that the spec is clear about the intent, and the current state of this PR is going to introduce a contradiction from what the spec says.

Quoting from [section 4.1](https://quicwg.org/base-drafts/draft-ietf-quic-recovery.html#section-4.1-1), "congestion control and round-trip time (RTT) measurement are unified across packet number spaces." As can be seen, persistent congestion by definition is a path-wide event. Not an event specific to the PN space in which a loss has been newly observed.

Based on that, my understanding is that consistently applying max_ack_delay exchanged by the Transport Parameter is the intent of [the formula found in in section 7.8](https://quicwg.org/base-drafts/draft-ietf-quic-recovery.html#section-7.8-3), and that the definition that currently exists in the WG draft is correct.

Reply:
I would move this comment out and to the definition of max_ack_delay further up in this section.

Reply:
It's also fine with me. And done in 5956eaa.